native/cocos/scene/SubModel.cpp

```diff
@@ -218,7 +220,16 @@ void SubModel::onPipelineStateChanged() {
 }
 
 void SubModel::onMacroPatchesStateChanged(const ccstd::vector<IMacroPatch> &patches) {
-    _patches = patches;
+    if (!patches.empty() && !patches.empty()) {
+        return;
+    }
+
+    ccstd::vector<IMacroPatch> tmp = patches;
+    std::sort(tmp.begin(), tmp.end(), IMacroPatch::compare);
+    if (std::equal(std::begin(tmp), std::end(tmp), std::begin(_patches))) {
```

Comment:
I think this feature is c++17 depend, it is better to pass `std::end(_patches)` too. For example
```c++
std::equal(std::begin(tmp), std::end(tmp), std::begin(_patches), std::end(_patches))
```

Reply:
> ```c++
> , std::end(_patches)
> ```

sure